online-shop/src/app/store/reducers/product.reducer.ts

```diff
@@ -0,0 +1,77 @@
+import { createReducer, on } from '@ngrx/store';
+import { addProduct, addProductFailure, addProductSuccess, deleteProduct, deleteProductSuccess, getProductById, getProductByIdFailure, getProductByIdSuccess, loadProducts, loadProductsFailure, loadProductsSuccess, updateProduct, updateProductFailure, updateProductSuccess } from '../actions/product.action';
+import { initialProductState } from '../state/product.state';
+  
+export const productReducer = createReducer(
+    // Supply the initial state
+    initialProductState,
+    on(addProduct, (state, { product }) => ({
+      ...state,
+      products: [...state.products, product],
+    })),
+    on(addProductSuccess, (state, {product}) => ({
+      ...state,
+      product,
+      error:"",
+      status: 'success'
+    })),
+    on(addProductFailure, (state, {error}) => ({
+      ...state,
+      error: error,
+      status: 'error'
+    })),
+
+    // Trigger loading the todosS
```

Comment:
Usually some developers prefer not having comments in within the code. The reason is that code should be self-explanatory.
Some short sayings about writing code comments:
- Writing and then maintaining comments is an expense.
- Your compiler doesn’t check your comments so there is no way to determine that comments are correct.